src/Lean/Elab/Tactic/BuiltinTactic.lean

```diff
@@ -379,6 +379,31 @@ private def getCaseGoals (tag : TSyntax ``binderIdent) : TacticM (MVarId × List
     setGoals (acc.toList ++ (← getGoals))
   | _ => throwUnsupportedSyntax
 
+@[builtinTactic «show»] def evalShow : Tactic
+  | `(tactic| show $e) => do
+    let mvarId :: mvarIds ← getUnsolvedGoals | throwNoGoalsToBeSolved
+    let s ← saveState
+    try
+      mvarId.withContext do
+        let mvarIdNew ← mkFreshExprMVar (← Term.elabType e)
+        mvarId.assign (← Term.ensureHasType (← mvarId.getType) mvarIdNew "'show' tactic failed")
```

Comment:
As you mentioned in the PR description, this will insert coercions if the types don't match.

I'm not sure if this such a good fit for a tactic that selects goals by type (as it is quite likely that it will just pick the first goal and insert a coercion).

Reply:
The coercion is not a subgoal, so I don't see how this would be a real concern unless both the source and target types are sufficiently known that the coercion can be fully synthesized. At least, it did not appear to be an issue in my tests.

Reply:
In that case, it sounds like it would suffice to call `synthesizeSyntheticMVarsNoPostponing` after the assignment, right?

Reply:
That might be a good idea anyhow, just to resolve things like `show ⟨1,2⟩ = _`.

I'm still not convinced by the automatic coercions.  Do you have a use case for that?

Reply:
What do you mean by "resolve" there? Will `show ⟨1,2⟩ = _` be rejected if we do that? (In particular, the `_` should not be an error nor should it produce a subgoal.)

As for automatic coercions, no it's not a big deal to lose them, but it would be more consistent with term mode `show`, which will perform coercions. (I would like tactics to behave the same as their term mode counterparts where reasonable.)